src/services/store-service.js

```diff
@@ -1,43 +1,37 @@
 (function() {
     'use strict';
 
-    angular.module('openfin.store', ['angular-storage'])
-        .factory('storeService', ['store', '$rootScope', function(store, $rootScope) {
-            var initialStocks = {
-                'AAPL': 0,
-                'MSFT': 1,
-                'TITN': 2,
-                'SNDK': 3,
-                'TSLA': 4
-            };
-
-            var favouriteStocks = store.get('stocks') || initialStocks;
-
-            function order(obj) {
-                var tuples = [];
-
-                for (var key in obj) {
-                    if (obj.hasOwnProperty(key)) {
-                        tuples.push([key, obj[key]]);
-                    }
+    angular.module('openfin.store', ['angular-storage', 'openfin.desktop'])
+        .factory('storeService', ['store', 'desktopService', '$rootScope', function(store, desktopService, $rootScope) {
+            var KEY_NAME = 'windows';
+            var initialStocks = [
+                {
+                    id: 'OpenFinD3FC',
+                    stocks: [
+                        'AAPL', 'MSFT', 'TITN', 'SNDK', 'TSLA'
+                    ],
+                    closed: false
                 }
+            ];
+
+            var storage = JSON.parse(store.get(KEY_NAME)) || initialStocks;
 
-                tuples = tuples.sort(function(a, b) {
-                    return a[1] - b[1];
-                });
+            function getWindowStore() {
+                var windowIndex = storage.map(function(window) {
+                    return window.id;
+                }).indexOf(desktopService.getCurrentWindow().name);
 
-                return tuples.map(function(x) {
-                    return x[0];
-                });
+                return windowIndex > -1 ? storage[windowIndex] : undefined; // TODO: undefined?
             }
 
             function save(stock) {
-                store.set('stocks', favouriteStocks);
+                store.set(KEY_NAME, JSON.stringify(storage));
```

Comment:
That's a very good point. In this current branch that won't be a problem has new windows can't be created yet, but in the (not too far.. !) future when they are I think I'll tackle this problem by promoting the store-service to be created initially by the overall parent, and shared with all the other windows. That will mean there is only one instance of it. I'll add some sort of comment to that effect.